From f26dbd9d9491333766ba383044064b1304127ac0 Mon Sep 17 00:00:00 2001
From: Venu Yeshala <vyeshala@codeaurora.org>
Date: Mon, 5 Jun 2017 10:55:00 +0530
Subject: msm: isp: camera: Avoid potential out of bound write

Check the validity of VFE interface parameter that is
provided from userspace.

Change-Id: I0ebb95c824ab3f832aaf500a6655829cca846c3d
Signed-off-by: Venu Yeshala <vyeshala@codeaurora.org>
---
 drivers/media/platform/msm/camera_v2/ispif/msm_ispif.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/media/platform/msm/camera_v2/ispif/msm_ispif.c b/drivers/media/platform/msm/camera_v2/ispif/msm_ispif.c
index e941ce00..97dfc773 100644
--- a/drivers/media/platform/msm/camera_v2/ispif/msm_ispif.c
+++ b/drivers/media/platform/msm/camera_v2/ispif/msm_ispif.c
@@ -1017,6 +1017,11 @@ static void msm_ispif_config_stereo(struct ispif_device *ispif,
 
 	for (i = 0; i < params->num; i++) {
 		vfe_intf = params->entries[i].vfe_intf;
+		if (!msm_ispif_is_intf_valid(ispif->csid_version, vfe_intf)) {
+			pr_err("%s: invalid interface type %d\n", __func__,
+				vfe_intf);
+			return;
+		}
 		if (params->entries[i].intftype == PIX0 &&
 			params->stereo_enable &&
 			params->right_entries[i].csid < CSID_MAX &&
-- 
cgit v1.1

